### PR TITLE
rs-project-setup

### PR DESCRIPTION
**Pull request by SakaRicky**
… start

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    "start": "next start",
+    "lint": "next lint"
+  },
+  "dependencies": {
+    "@material-ui/core": "^4.12.3",
+    "@material-ui/icons": "^4.11.2",
+    "next": "11.1.2",
+    "react": "17.0.2",
+    "react-dom": "17.0.2",
+    "sass": "^1.43.2"
```
Comment (lkuz) [inline package.json:17]: No issue here but you should keep in mind that there exists Dart Sass - https://sass-lang.com/dart-sass

File: pages/_app.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import '../styles/globals.scss'
```
Comment (lkuz) [inline pages/_app.tsx:1]: use absolute path's instead of relative (e.g. 'styles/globals.scss')

File: next.config.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/** @type {import('next').NextConfig} */
+const path = require('path')
+
+module.exports = {
+  reactStrictMode: true,
+  sassOptions: {
+    includePaths: [path.join(__dirname, 'styles')],
```
Comment (lkuz) [inline next.config.js:7]: Is it necessary to include that path in order do load the sass files properly? Doesn't next handle this OotB?

File: pages/api/hello.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+// Next.js API route support: https://nextjs.org/docs/api-routes/introduction
+import type { NextApiRequest, NextApiResponse } from 'next'
+
+type Data = {
+  name: string
+}
```
Comment (lkuz) [inline pages/api/hello.ts:6]: Is this type definition necessary?

File: package.json (hunk 1 above)
Comment (SakaRicky) [reply package.json:17]: Thanks for this. I see there are 3 versions of Dart, command line, Dart library and Dart JavaScript Library. Which one do you recommend?

File: next.config.js (hunk 3 above)
Comment (lkuz) [reply next.config.js:7]: In case you wanted to expose all styles globally, check out _app: https://nextjs.org/docs/advanced-features/custom-app

**Review by lkuz — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (lkuz) [reply package.json:17]: Whenever you feel like you have seen everything in sass - that's the moment you should think about that.

Just good to know for now

Comment (SakaRicky) [reply package.json:17]: Ok thanks

File: pages/_app.tsx (hunk 2 above)
Comment (SakaRicky) [reply pages/_app.tsx:1]: How to know when to copy relative paths or absolute paths?
Or I should always copy absolute paths

File: next.config.js (hunk 3 above)
Comment (SakaRicky) [reply next.config.js:7]: Oh I corrected it. I copied that code from the official doc on using sass

File: pages/_app.tsx (hunk 2 above)
Comment (lkuz) [reply pages/_app.tsx:1]: You can read about the pros and cons of absolute and relative imports here: [link]( https://beqode.com/blog/absolute-vs-relative-import-paths-nodejs)

File: next.config.js (hunk 3 above)
Comment (lkuz) [reply next.config.js:7]: No problem, just read more careful

**Pull request closed without merging**